common/djangoapps/edxmako/tests.py

```diff
@@ -41,6 +41,29 @@ def test_marketing_link(self):
             link = marketing_link('ABOUT')
             self.assertEquals(link, expected_link)
 
+    # @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
```

Comment:
I think it might have been that the ddt.data decorator called reverse before it got into the actual  test_marketing_link_external_courses_url function, which would explain why in the log, the test couldn't find the "courses" key.